Coreclr build errors related to "enum class" usage.

Can't build coreclr after commit https://github.com/dotnet/runtime/commit/b8591b4a9ce88f98ff6aba93241f23e1512c01fa

Ubuntu 18.04, clang-9, cmake-3.16
Steps to reproduce:
```
$ sudo ./eng/common/cross/build-rootfs.sh armel tizen
$ ./src/coreclr/build.sh cross -armel -Release -clang9 -skipmanagedtools -skiprestore -skipnuget -skipcrossgen -msbuildonunsupportedplatform -cmakeargs "-DFEATURE_NGEN_RELOCS_OPTIMIZATIONS=true -DFEATURE_PREJIT=true -DARM_SOFTFP=true -DFEATURE_ENABLE_NO_ADDRESS_SPACE_RANDOMIZATION=true -DCLR_ADDITIONAL_LINKER_FLAGS=-Wl,-z,relro"
```
Error log:
```
Scanning dependencies of target daccess
[ 99%] Building CXX object src/debug/daccess/CMakeFiles/daccess.dir/daccess.cpp.o
[ 99%] Building CXX object src/debug/daccess/CMakeFiles/daccess.dir/enummem.cpp.o
[ 99%] Building CXX object src/debug/daccess/CMakeFiles/daccess.dir/nidump.cpp.o
/home/viewizard/Desktop/runtime/src/coreclr/src/debug/daccess/nidump.cpp:8840:28: error: invalid operands to binary expression ('ReadyToRunSectionType' and 'DWORD' (aka 'unsigned int'))
        if (pSection->Type == type)
            ~~~~~~~~~~~~~~ ^  ~~~~
/home/viewizard/Desktop/runtime/src/coreclr/src/inc/daccess.h:1647:13: note: candidate template ignored: could not match '__GlobalPtr<type-parameter-0-0, type-parameter-0-1>' against 'ReadyToRunSectionType'
inline bool operator==(const __GlobalPtr<acc_type, store_type>& gptr,
            ^
/home/viewizard/Desktop/runtime/src/coreclr/src/inc/daccess.h:1661:13: note: candidate template ignored: could not match '__GlobalPtr<type-parameter-0-0, type-parameter-0-1>' against 'unsigned int'
inline bool operator==(acc_type host,
            ^
/home/viewizard/Desktop/runtime/src/coreclr/src/debug/daccess/nidump.cpp:8855:73: error: cannot initialize a parameter of type 'DWORD' (aka 'unsigned int') with an rvalue of type 'ReadyToRunSectionType'
    IMAGE_DATA_DIRECTORY * pRuntimeFunctionsDir = FindReadyToRunSection(ReadyToRunSectionType::RuntimeFunctions);
                                                                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/viewizard/Desktop/runtime/src/coreclr/src/debug/daccess/nidump.cpp:8831:71: note: passing argument to parameter 'type' here
IMAGE_DATA_DIRECTORY * NativeImageDumper::FindReadyToRunSection(DWORD type)
                                                                      ^
/home/viewizard/Desktop/runtime/src/coreclr/src/debug/daccess/nidump.cpp:8866:68: error: cannot initialize a parameter of type 'DWORD' (aka 'unsigned int') with an rvalue of type 'ReadyToRunSectionType'
    IMAGE_DATA_DIRECTORY * pEntryPointsDir = FindReadyToRunSection(ReadyToRunSectionType::MethodDefEntryPoints);
                                                                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/viewizard/Desktop/runtime/src/coreclr/src/debug/daccess/nidump.cpp:8831:71: note: passing argument to parameter 'type' here
IMAGE_DATA_DIRECTORY * NativeImageDumper::FindReadyToRunSection(DWORD type)
                                                                      ^
3 errors generated.
src/debug/daccess/CMakeFiles/daccess.dir/build.make:231: recipe for target 'src/debug/daccess/CMakeFiles/daccess.dir/nidump.cpp.o' failed
make[2]: *** [src/debug/daccess/CMakeFiles/daccess.dir/nidump.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
CMakeFiles/Makefile2:2928: recipe for target 'src/debug/daccess/CMakeFiles/daccess.dir/all' failed
make[1]: *** [src/debug/daccess/CMakeFiles/daccess.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2
Failed to build "CoreCLR component".
```
Probably, ```enum class``` should use explicit cast to apropriate type?

--- a/src/coreclr/src/debug/daccess/nidump.cpp
+++ b/src/coreclr/src/debug/daccess/nidump.cpp
@@ -8828,7 +8828,7 @@ StandardEntryDisplay:
 }
 
 #ifdef FEATURE_READYTORUN
-IMAGE_DATA_DIRECTORY * NativeImageDumper::FindReadyToRunSection(DWORD type)
+IMAGE_DATA_DIRECTORY * NativeImageDumper::FindReadyToRunSection(ReadyToRunSectionType type)
 {
     PTR_READYTORUN_SECTION pSections = dac_cast<PTR_READYTORUN_SECTION>(dac_cast<TADDR>(m_pReadyToRunHeader) + sizeof(READYTORUN_HEADER));
     for (DWORD i = 0; i < m_pReadyToRunHeader->NumberOfSections; i++)

--- a/src/coreclr/src/debug/daccess/nidump.h
+++ b/src/coreclr/src/debug/daccess/nidump.h
@@ -244,7 +244,7 @@ private:
     NativeFormat::NativeReader  m_nativeReader;
     NativeFormat::NativeArray   m_methodDefEntryPoints;
 
-    IMAGE_DATA_DIRECTORY * FindReadyToRunSection(DWORD type);
+    IMAGE_DATA_DIRECTORY * FindReadyToRunSection(ReadyToRunSectionType type);
 
 public:
     void DumpReadyToRun();

